astropy/visualization/stretch.py

```diff
@@ -80,21 +85,29 @@ def inverse(self):
 
 class LinearStretch(BaseStretch):
     """
-    A linear stretch.
+    A linear stretch with a slope and offset.
 
     The stretch is given by:
 
     .. math::
-        y = x
+        y = ax + b
     """
 
+    def __init__(self, a=1, b=0):
+        super().__init__()
+        self.slope = a
+        self.intercept = b
+
     def __call__(self, values, clip=True, out=None):
-        return _prepare(values, clip=clip, out=out)
+        values = _prepare(values, clip=clip, out=out)
+        np.multiply(values, self.slope, out=values)
+        np.add(values, self.intercept, out=values)
```

Comment:
It might be worth doing each of these operations only if the slope is not 1 and the intercept is not 0, to avoid performance getting worse in the default case.